aea/agent_loop.py

```diff
@@ -231,13 +99,7 @@ def __init__(self, agent: "AEA", loop: AbstractEventLoop = None):
         super().__init__(agent)
         self._agent: "AEA" = self._agent
 
-        try:
-            self._loop = loop or asyncio.get_event_loop()
-            assert not self._loop.is_closed()
-            assert not self._loop.is_running()
-        except (RuntimeError, AssertionError):
-            self._loop = asyncio.new_event_loop()
-            asyncio.set_event_loop(self._loop)
+        self._loop: AbstractEventLoop = ensure_loop(loop)
```

Comment:
`get_loop` would maybe be more explicit?

Reply:
I will mark as TODO for this part, it's unclean a bit how to pass loop and who will construct it.